### ISSUE TITLE
openshift_logging and openshift_metrics roles pulling the wrong image tags for 3.5

### ISSUE THREAD

**Issue by etsauer** [openshift/openshift-ansible#3931]
#### Description

When installing OCP 3.5, the openshift_logging and openshift_metrics roles attempt to pull the tag `:3.5` for the images, resulting in `ImagePullBackoff` and `Error` states in all of the pods.

It looks like the intended tag to pull is either `3.5.0` or `v3.5`:

```
$ curl -X GET -s -L https://registry.access.redhat.com/v2/openshift3/metrics-cassandra/tags/list | python -m json.tool | grep '3.5'
        "3.5.0-7",
        "v3.5",
        "3.5.0",
```

Some sample error messages:

```
  10h           4m              128     {kubelet app-node-0.andy.casl.rht-labs.com}     spec.containers{hawkular-cassandra-1}   Warning         Failed          Failed to pull image "registry.access.redhat.com/openshift3/metrics-cassandra:3.5": image pull failed for registry.access.redhat.com/openshift3/metrics-cassandra:3.5, this may be because there are no credentials on this request.  details: (error parsing HTTP 404 response body: invalid character 'F' looking for beginning of value: "File not found.\"")
  11h           2m              133     {kubelet app-node-0.andy.casl.rht-labs.com}     spec.containers{hawkular-cassandra-1}   Warning         Failed          Failed to pull image "registry.access.redhat.com/openshift3/metrics-cassandra:3.5": image pull failed for registry.access.redhat.com/openshift3/metrics-cassandra:3.5, this may be because there are no credentials on this request.  details: (error parsing HTTP 404 response body: invalid character 'F' looking for beginning of value: "File not found.\"")
  11h           4m              133     {kubelet app-node-0.andy.casl.rht-labs.com}     spec.containers{heapster}       Warning         Failed          Failed to pull image "registry.access.redhat.com/openshift3/metrics-heapster:3.5": image pull failed for registry.access.redhat.com/openshift3/metrics-heapster:3.5, this may be because there are no credentials on this request.  details: (error parsing HTTP 404 response body: invalid character 'F' looking for beginning of value: "File not found.\"")
  11h           3m              132     {kubelet infranode-0.andy.casl.rht-labs.com}    spec.containers{hawkular-metrics}       Warning         Failed          Failed to pull image "registry.access.redhat.com/openshift3/metrics-hawkular-metrics:3.5": image pull failed for registry.access.redhat.com/openshift3/metrics-hawkular-metrics:3.5, this may be because there are no credentials on this request.  details: (error parsing HTTP 404 response body: invalid character 'F' looking for beginning of value: "File not found.\"")

```

##### Version
```
$ 
ansible --version
ansible 2.3.0.0
  config file = /root/.ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Nov  6 2016, 00:28:07) [GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]
$ git describe
openshift-ansible-3.6.23-1-10-g298f1aa
```

##### Steps To Reproduce
1. Add the following to ansible inventory
```
openshift_deployment_type: openshift-enterprise
openshift_release: v3.5

openshift_hosted_logging_deploy: true
openshift_hosted_logging_storage_kind: dynamic

openshift_hosted_metrics_deploy: true
openshift_hosted_metrics_storage_kind: dynamic
```
2. run `playbooks/byo/config.yml`

##### Expected Results
Metrics and Logging pods are `Running`

##### Observed Results
Describe what is actually happening.

```
$ oc get pods -n logging
NAME                           READY     STATUS             RESTARTS   AGE
logging-curator-1-deploy       0/1       Error              0          11h
logging-es-7s5kkpo6-1-deploy   0/1       Error              0          11h
logging-fluentd-35fdx          0/1       ImagePullBackOff   0          11h
logging-fluentd-5zhk8          0/1       ImagePullBackOff   0          11h
logging-fluentd-n4nss          0/1       ImagePullBackOff   0          11h
logging-kibana-1-deploy        0/1       Error              0          11h

$ oc get pods -n openshift-infra
NAME                         READY     STATUS             RESTARTS   AGE
hawkular-cassandra-1-qdxh7   0/1       ImagePullBackOff   0          11h
hawkular-metrics-xgrqz       0/1       ImagePullBackOff   0          11h
heapster-74vmt               0/1       ErrImagePull       0          11h
```

For long output or logs, consider using a [gist](https://gist.github.com/)

@ewolinetz

**Comment by etsauer** [comment]
so it looks like the intention here is to pull the tag as specified in `openshift_release`, and then default to `3.5.0`.. 

```
$ cat roles/openshift_logging/vars/openshift-enterprise.yml 
---
__openshift_logging_image_prefix: "{{ openshift_hosted_logging_deployer_prefix | default('registry.access.redhat.com/openshift3/') }}"
__openshift_logging_image_version: "{{ openshift_hosted_logging_deployer_version | default(openshift_release | default ('3.5.0') ) }}"
```

But somehow it's pulling `3.5` as the value of `openshift_release` instead of `v3.5`?

**Comment by sdodson** [comment]
This was fixed on release-1.5 yesterday. Not sure if the master PR merged
yet.

On Apr 14, 2017 8:55 AM, "Eric Sauer" <notifications@github.com> wrote:

so it looks like the intention here is to pull the tag as specified in
openshift_release, and then default to 3.5.0..

$ cat roles/openshift_logging/vars/openshift-enterprise.yml
---
__openshift_logging_image_prefix: "{{
openshift_hosted_logging_deployer_prefix |
default('registry.access.redhat.com/openshift3/') }}"
__openshift_logging_image_version: "{{
openshift_hosted_logging_deployer_version | default(openshift_release
| default ('3.5.0') ) }}"

But somehow it's pulling 3.5 as the value of openshift_release instead of
v3.5?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/openshift-ansible/issues/3931#issuecomment-294148980>,
or mute the thread
